### PR TITLE
added ability to get region specific bucket naming

### PR DESCRIPTION
**Pull request by ndcampbell**
When retrieving a bucket name, can specify if it should return a region specific bucket name or not. 

Will added unit tests

### CODE REVIEW

File: src/gogoutils/generator.py [hunk 1]
```diff
@@ -113,14 +113,28 @@ def dns(self):
 
         return dns
 
-    def s3_app_bucket(self):
-        """Generates s3 application bucket name."""
-        s3_app_bucket = self.format['s3_app_bucket'].format(**self.data)
+    def s3_app_bucket(self, include_region=True):
```
Comment (sijis) [inline src/gogoutils/generator.py]: Default should be `False`, to keep existing behavior.

File: src/gogoutils/generator.py [hunk 2]
```diff
@@ -113,14 +113,28 @@ def dns(self):
...
+        if include_region:
+            s3_app_bucket = self.format['s3_app_region_bucket'].format(**self.data)
+        else:
+            s3_app_bucket = self.format['s3_app_bucket'].format(**self.data)
         return s3_app_bucket
 
-    def shared_s3_app_bucket(self):
-        """Generates shared s3 application bucket name."""
-        shared_s3_app_bucket = self.format['shared_s3_app_bucket'].format(**self.data)
+    def shared_s3_app_bucket(self, include_region=True):
```
Comment (sijis) [inline src/gogoutils/generator.py]: Default should be `False`, to keep existing behavior.

**Comment by ndcampbell** [comment]
Unit tests and docs are fixed @sijis

File: src/gogoutils/generator.py [hunk 3]
```diff
@@ -113,14 +113,28 @@ def dns(self):
 
         return dns
 
-    def s3_app_bucket(self):
-        """Generates s3 application bucket name."""
-        s3_app_bucket = self.format['s3_app_bucket'].format(**self.data)
+    def s3_app_bucket(self, include_region=False):
+        """Generates s3 application bucket name.
+
+        Args;
```
Comment (sijis) [inline src/gogoutils/generator.py]: Colon.

File: src/gogoutils/generator.py [hunk 4]
```diff
@@ -113,14 +113,28 @@ def dns(self):
...
+            s3_app_bucket = self.format['s3_app_bucket'].format(**self.data)
         return s3_app_bucket
 
-    def shared_s3_app_bucket(self):
-        """Generates shared s3 application bucket name."""
-        shared_s3_app_bucket = self.format['shared_s3_app_bucket'].format(**self.data)
+    def shared_s3_app_bucket(self, include_region=False):
+        """Generates shared s3 application bucket name.
+
+        Args;
```
Comment (sijis) [inline src/gogoutils/generator.py]: Colon.

File: src/gogoutils/generator.py [hunk 5]
```diff
@@ -113,14 +113,28 @@ def dns(self):
...
         return dns
 
-    def s3_app_bucket(self):
-        """Generates s3 application bucket name."""
-        s3_app_bucket = self.format['s3_app_bucket'].format(**self.data)
+    def s3_app_bucket(self, include_region=False):
+        """Generates s3 application bucket name.
+
+        Args;
+            include_region (bool): If to include region in the name generation.
```
Comment (sijis) [inline src/gogoutils/generator.py]: Removing `If to ` makes that read better, i think.

**Pull request merged**